multipathd/uxclnt.c

```diff
@@ -70,20 +69,20 @@ static int need_quit(char *str, size_t len)
  */
 static void process(int fd, unsigned int timeout)
 {
-	char *line;
+	char line[256];
 	char *reply;
+	unsigned int i;
 	int ret;
+	char prompt[] = "multipathd> ";
 
 	cli_init();
-	rl_readline_name = "multipathd";
-	rl_completion_entry_function = key_generator;
-	while ((line = readline("multipathd> "))) {
+	for (i = 0; i < strlen(prompt); i++)
+		fputc(prompt[i], stdout);
+	while (fgets(line, 256, stdin)) {
```

Comment:
We'd need some error handling here, especially for overflow, and we should strip training newline. I'd prefer using `getline()`, actually.

Reply:
Oh, sure. The really was a quick-and-dirty hack, and fgets() was the first function which came to my mind. getline() is fine, too.